masquerade-core/src/main/java/com/flipkart/masquerade/processor/type/EnumOverrideProcessor.java

```diff
@@ -14,36 +14,42 @@
  * limitations under the License.
  */
 
-package com.flipkart.masquerade.processor;
+package com.flipkart.masquerade.processor.type;
 
 import com.flipkart.masquerade.Configuration;
+import com.flipkart.masquerade.processor.BaseOverrideProcessor;
 import com.flipkart.masquerade.rule.Rule;
 import com.squareup.javapoet.MethodSpec;
 import com.squareup.javapoet.TypeSpec;
 
-import static com.flipkart.masquerade.util.Helper.getNoOpImplementationName;
+import static com.flipkart.masquerade.util.Helper.getEnumImplementationName;
+import static com.flipkart.masquerade.util.Strings.OBJECT_PARAMETER;
+import static com.flipkart.masquerade.util.Strings.QUOTES;
 
 /**
- * Processor that creates a No-Op implementation class for a Mask interface
- * <p />
- * Created by shrey.garg on 27/05/17.
+ * Created by shrey.garg on 09/07/17.
  */
-public class NoOpOverrideProcessor extends BaseOverrideProcessor {
+public class EnumOverrideProcessor extends BaseOverrideProcessor {
     /**
      * @param configuration Configuration for the current processing cycle
      * @param cloakBuilder  Entry class under construction for the cycle
      */
-    public NoOpOverrideProcessor(Configuration configuration, TypeSpec.Builder cloakBuilder) {
+    public EnumOverrideProcessor(Configuration configuration, TypeSpec.Builder cloakBuilder) {
         super(configuration, cloakBuilder);
     }
 
     /**
      * @param rule The rule which is being processed
-     * @return A fully constructed TypeSpec object for the no op implementation
+     * @return A fully constructed TypeSpec object for the enum implementation
      */
     public TypeSpec createOverride(Rule rule) {
-        String implName = getNoOpImplementationName(rule);
+        String implName = getEnumImplementationName(rule);
         MethodSpec.Builder methodBuilder = generateOverrideMethod(rule, Object.class);
+
+        if (configuration.isNativeSerializationEnabled()) {
+            methodBuilder.addStatement("return $S + String.valueOf($L) + $S", QUOTES, OBJECT_PARAMETER, QUOTES);
+        }
```

Comment:
This is a simple enum serialization case. JsonFormat and JsonValue can be used on enums to control how to serialize enums. We don't have such use cases to support on v1?